### PR TITLE
ARROW-241: Checking list-types in schema support raises an error

### PR DESCRIPTION
**Pull request by lazargugleta**
[JIRA TICKET](https://jira.mongodb.org/browse/ARROW-241)

### CODE REVIEW

**Comment by Jibola** [comment]
Thanks for supplying this fix @lazargugleta!

One last thing we'd need is for you to add a test case for this function. This ensures that we will not see this error again.

**Review by Jibola — CHANGES_REQUESTED** [review]
See above.

**Comment by lazargugleta** [comment]
@Jibola 
Hey thanks for the quick reply, I added the test
Hopefully in the correct place. Let me know If there is anything else needed.

File: bindings/python/pymongoarrow/types.py [hunk 1]
```diff
@@ -288,7 +288,7 @@ def _normalize_typeid(typeid, field_name):
             fields.append((sub_field_name, _normalize_typeid(sub_typeid, sub_field_name)))
         return struct(fields)
     if isinstance(typeid, list):
-        return list_(_normalize_typeid(type(typeid[0]), "0"))
+        return list_(_normalize_typeid(typeid[0], "0"))
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/types.py:291]: What happens if the provided list has >1 value in it? Should we error or silently ignore it?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bindings/python/pymongoarrow/types.py (hunk 1 above)
Comment (lazargugleta) [reply bindings/python/pymongoarrow/types.py:291]: Good point, I would say we could raise with a message that the struct should be provided.

Comment (ShaneHarvey) [reply bindings/python/pymongoarrow/types.py:291]: That sounds reasonable to me. A ValueError would be appropriate.

Comment (lazargugleta) [reply bindings/python/pymongoarrow/types.py:294]: Done

File: bindings/python/pymongoarrow/types.py [hunk 2]
```diff
@@ -288,7 +288,10 @@ def _normalize_typeid(typeid, field_name):
             fields.append((sub_field_name, _normalize_typeid(sub_typeid, sub_field_name)))
         return struct(fields)
     if isinstance(typeid, list):
-        return list_(_normalize_typeid(type(typeid[0]), "0"))
+        if len(typeid) == 0 or len(typeid) > 1:
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/types.py:291]: Suggest `if len(typeid) != 1`

File: bindings/python/pymongoarrow/types.py [hunk 3]
```diff
@@ -288,7 +288,10 @@ def _normalize_typeid(typeid, field_name):
...
+            msg = "Please provide a one datatype or struct in case of multiple datatypes in a list"
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/types.py:292]: Suggest: `f"list field in schema must contain exactly one element, not {len(typeid)}"`

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bindings/python/pymongoarrow/types.py (hunk 2 above)
Comment (lazargugleta) [reply bindings/python/pymongoarrow/types.py]: Accept

File: bindings/python/pymongoarrow/types.py (hunk 3 above)
Comment (lazargugleta) [reply bindings/python/pymongoarrow/types.py]: Accept

**Comment by ShaneHarvey** [comment]
LGTM, scheduling the tests.

**Comment by ShaneHarvey** [comment]
Why remove test_py_empty_list_raises?

**Comment by lazargugleta** [comment]
> Why remove test_py_empty_list_raises?

It tests the same case.
Only good thing about leaving it in is that is shows intended behaviour.
Should I leave it in?

**Comment by ShaneHarvey** [comment]
Yeah could you add it back? It's nice to have as an extra regression test. Also our pre-commit checks are failing. You will need to install and run:
```
pip install pre-commit
pre-commit run --all-files
```

**Comment by lazargugleta** [comment]
> Yeah could you add it back? It's nice to have as an extra regression test. Also our pre-commit checks are failing. You will need to install and run:
> 
> ```
> pip install pre-commit
> pre-commit run --all-files
> ```

Thanks, test added back + reformatted code according to formating

**Review by ShaneHarvey — APPROVED** [review]
Thanks!

**Review by Jibola — APPROVED** [review]

**Pull request merged**